examples/ota-requestor-app/linux/main.cpp

```diff
@@ -189,5 +189,23 @@ int main(int argc, char * argv[])
 {
     VerifyOrDie(ChipLinuxAppInit(argc, argv, &cmdLineOptions) == 0);
     ChipLinuxAppMainLoop();
+
+    // If the event loop had been stopped due to an update being applied, boot into the new image
+    if (gRequestorCore.GetCurrentUpdateState() == OTARequestor::OTAUpdateStateEnum::kApplying)
+    {
+        if (kMaxFilePathSize <= strlen(kImageExecPath))
+        {
+            ChipLogError(SoftwareUpdate, "Buffer too small for the new image file path: %s", kImageExecPath);
+            return -1;
+        }
+
+        char execFilePathBuf[kMaxFilePathSize];
+        strncpy(execFilePathBuf, kImageExecPath, strlen(kImageExecPath));
```

Comment:
Why does it need to be copied? Can't `kImageExecPath` be assigned directly?

Reply:
So the issue is that `kImageExecPath` is const and argv[0] takes in `char *`. I guess I can change `kImageExecPath` to non-const to do this assignment directly